### PR TITLE
Update tests, fix many things, and add role role.

### PR DESCRIPTION
**Pull request by sean-m-sullivan**
# What does this PR do?
This adds 83% test coverage to every option feasible to test, making sure all of those are done with at least a role where possible. There are about 22/125 options we are unable to test due to proxy, certs, and other things, but this brings up the testing considerably, and fixes problems where found.

minor_changes:
  - Updated ah_collection_upload to exit properly.
  - Updated testing coverage.
  - Updated ah_role to not require permisions in order to allow for deletion.
  - Added ansible role 'role' for creating roles of group permsions.
  - Fixed issue with ah_ee_repository working on newer versions.
  - Fixed issue with ah_namespace delete
  - ah_ee_repository include/exclude tags are no longer mutually exclusive
  - added better error catching to ah_ee_registery_index/sync to handle when a registery is not found.
breaking_changes:
  - Introduces a `sync` option to the ee_repository_sync role on the `ah_ee_repositories` variable which by default is false and which is required true to sync the repository.
  - Removed ah_ee_registry new name option. This affected underlying images indexed from the registery leaving them abandoned, and subsequent indexes of the renamned registery would fail.

# How should this be tested?
This should be tested with the testing that we have in addition to the testing added.

# Is there a relevant Issue open for this?
#184 
#183 
#182 
#138
#135

### CODE REVIEW

**Comment by sean-m-sullivan** [comment]
Because of the breadth of the the changes, I want Tom to review this before we merge it.

**Comment by sean-m-sullivan** [comment]
Also because of added variables aren't allowed in PR's, it will fail, however here is the same commit passing: https://github.com/sean-m-sullivan/ah_configuration/actions/runs/3977449123

**Review by silvinux — COMMENTED** [review]
Awesome job sean!!!. Looks awesome!

File: plugins/modules/ah_role.py [hunk 1]
```diff
@@ -159,7 +158,7 @@ def main():
     argument_spec = dict(
         name=dict(required=True),
         description=dict(),
-        perms=dict(type="list", elements="str", required=True),
+        perms=dict(type="list", elements="str"),
```
Comment (Tompage1994) [inline plugins/modules/ah_role.py:161]: If there is no required here, do we not want to default this to `[]`?

**Review by Tompage1994 — APPROVED** [review]
LGTM - Though theres one question in there that I'm not sure about, happy if you don't think it needs to be fixed though

File: plugins/modules/ah_role.py (hunk 1 above)
Comment (Tompage1994) [reply plugins/modules/ah_role.py:161]: And upon checking, yes, this does need to be defaulted to `[]` otherwise it fails

Comment (sean-m-sullivan) [reply plugins/modules/ah_role.py:161]: Need to do a test, but going to fix it.

Comment (sean-m-sullivan) [reply plugins/modules/ah_role.py:161]: ok fixed and tested, also found another error, but for now its just only set the perms defined, we could add some fancy add to, etc, but made it static like the roles in awx.awx.

**Pull request merged**